stellargraph/layer/graphsage.py

```diff
@@ -690,12 +690,12 @@ def node_model(self, flatten_output=False):
         # Output from GraphSAGE model
         x_out = self(x_inp)
 
-        if flatten_output:
-            x_out = Reshape((-1,))(x_out)
+        if K.int_shape(x_out)[1] == 1:
```

Comment:
Inconsistency with HinSAGE: in HinSAGE, removal of the "neighbourhood" dimension is done in `HinSAGE.__call__()` method (lines 382-385 in `hinsage.py`), while here it is done in `GraphSAGE.node_model()` method. Perhaps this should be moved to `GraphSAGE.__call__()` as well.

Also, a comment should be added to this operation, similar to that in HinSAGE